### ISSUE TITLE
Training result reporting formatting is strange

### ISSUE THREAD

**Issue by kparichay** [nnstreamer/nntrainer#579]
Training result reporting is done in a very strange format. Further, this format is not the same across OS as well : 
Below is the output of training transfer learning application (Draw classification) using C-API on ubuntu:
```bash
13:42:05 1z [:~/Working/nnsuite/nntrainer_pk.git/build] cc_api+ ± ./Applications/TransferLearning/Draw_Classification/jni/nntrainer_training ../Applications/TransferLearning/Draw_Classification/res/Training.ini ../Applications/TransferLearning/Draw_Classification/res
#1/1000 - Training Loss: 1.14831  ( Training Loss: 0.961607 )
#2/1000 - Training Loss: 1.1346]  ( Training Loss: 0.96903 ))
#3/1000 - Training Loss: 1.12772  ( Training Loss: 0.977494 )
#4/1000 - Training Loss: 1.12354  ( Training Loss: 0.983625 )
#5/1000 - Training Loss: 1.1206]  ( Training Loss: 0.987452 )
#6/1000 - Training Loss: 1.11827  ( Training Loss: 0.98962 ))
#7/1000 - Training Loss: 1.11627  ( Training Loss: 0.990714 )
#8/1000 - Training Loss: 1.11445  ( Training Loss: 0.991145 )
#9/1000 - Training Loss: 1.11272  ( Training Loss: 0.99118 ))
#10/1000 - Training Loss: 1.11104  ( Training Loss: 0.99098 ))
#11/1000 - Training Loss: 1.10938  ( Training Loss: 0.990642 )
...
#994/1000 - Training Loss: 0.039463 ( Training Loss: 0.114078 )))
#995/1000 - Training Loss: 0.0393834( Training Loss: 0.113887 )))
#996/1000 - Training Loss: 0.0393041( Training Loss: 0.113698 )))
#997/1000 - Training Loss: 0.039225 ( Training Loss: 0.113508 )))
#998/1000 - Training Loss: 0.0391462( Training Loss: 0.113319 ))
#999/1000 - Training Loss: 0.0390677( Training Loss: 0.11313 ) ))
#1000/1000 - Training Loss: 0.0389895( Training Loss: 0.112942 )))
```
This shows two training losses for some reason. Further, the brackets are quite strange.

Below is the output of training transfer learning application (Draw classification) using C-API on Tizen:
```bash
[   74s] + ./Applications/TransferLearning/Draw_Classification/jni/nntrainer_training ../Applications/TransferLearning/Draw_Classification/res/Training.ini ../Applications/TransferLearning/Draw_Classification/res
[   74s] ** Message: 04:43:10.149: nnapi = 1, accl = cpu
#1/1000 - Training Loss: 1.14831  ( Training Loss: 0.961607 ).53722 )
#2/1000 - Training Loss: 1.1346]  ( Training Loss: 0.96903 )).40274 )
#3/1000 - Training Loss: 1.12772  ( Training Loss: 0.977494 ).33276 )
#4/1000 - Training Loss: 1.12354  ( Training Loss: 0.983625 ).29503 )
#5/1000 - Training Loss: 1.1206]  ( Training Loss: 0.987452 ).27381 )
#6/1000 - Training Loss: 1.11827  ( Training Loss: 0.98962 )).26111 )
#7/1000 - Training Loss: 1.11627  ( Training Loss: 0.990714 ).25285 )
#8/1000 - Training Loss: 1.11445  ( Training Loss: 0.991145 ).24693 )
#9/1000 - Training Loss: 1.11272  ( Training Loss: 0.99118 )).24225 )
#10/1000 - Training Loss: 1.11104  ( Training Loss: 0.99098 )).23822 )
#11/1000 - Training Loss: 1.10938  ( Training Loss: 0.990642 ).23455 )
...
#994/1000 - Training Loss: 0.039463 ( Training Loss: 0.114078 )))123612 )
#995/1000 - Training Loss: 0.0393834( Training Loss: 0.113888 )))123409 )
#996/1000 - Training Loss: 0.0393041( Training Loss: 0.113698 )))123203 )
#997/1000 - Training Loss: 0.0392251( Training Loss: 0.113508 )))123001 )
#998/1000 - Training Loss: 0.0391462( Training Loss: 0.113319 ))01228 )
#999/1000 - Training Loss: 0.0390677( Training Loss: 0.11313 ) ))122597 )
#1000/1000 - Training Loss: 0.0389894( Training Loss: 0.112942 )))122397 )
```
This is even strange with that extra value printed in the end.

**Comment by zhoonit** [comment]
I noticed samething in the CustomShortcut.  I think the problem is either with buffer or stream as it prints out with `\r`

**Comment by zhoonit** [comment]
FYI) What I am getting from customshortcut is `#11/20 [ =                    ] 0% ( Training Loss: 0.277#11/20 - Training Loss: 0.277974 >> [ Accuracy: 100% - Validation Loss : 0.251338 ]`

**Comment by kparichay** [comment]
> I noticed samething in the CustomShortcut. I think the problem is either with buffer or stream as it prints out with `\r`

Something related to this or `std::cout.flush()` as it plays with the cout buffer. There is no other candidate as this is single-threaded code.
